code/builders/builder-vite/src/codegen-modern-iframe-script.ts

```diff
@@ -90,6 +91,10 @@ export async function generateModernIframeScriptCodeFromPreviews(options: {
 
     return dedent`
     if (import.meta.hot) {
+      import.meta.hot.on('vite:afterUpdate', () => {
+        window.__STORYBOOK_PREVIEW__.channel.emit('${STORY_HOT_UPDATED}');
+      });
```

Comment:
Have we tested this with react fast refresh?
Because fast refresh catches the HMR event and does not let it propagate further.

Reply:
@ndelangen How would I do that? Do you have a suggestion on how to detect fast refresh?